File: C/Algorithms/145.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

void traverse(struct TreeNode *node, int **values, int *length, int *size) {
  // Go left first.
  if (node->left)
    traverse(node->left, values, length, size);

  if (node->right)
    traverse(node->right, values, length, size);

  if ((*length + 1) >= *size) {
    *size *= 2;
    *values = realloc(*values, *size * sizeof(*values));
  }
  (*values)[(*length)++] = node->val;
}

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int *postorderTraversal(struct TreeNode *root, int *returnSize) {
  *returnSize = 0;
  if (!root) {
    return NULL;
  }

  int size = 1;
  int *values = malloc(size * sizeof(int));
  traverse(root, &values, returnSize, &size);

  return values;
}

File: Rust/Algorithms/145.rs
// Definition for a binary tree node.
// #[derive(Debug, PartialEq, Eq)]
// pub struct TreeNode {
//   pub val: i32,
//   pub left: Option<Rc<RefCell<TreeNode>>>,
//   pub right: Option<Rc<RefCell<TreeNode>>>,
// }
// 
// impl TreeNode {
//   #[inline]
//   pub fn new(val: i32) -> Self {
//     TreeNode {
//       val,
//       left: None,
//       right: None
//     }
//   }
// }
use std::rc::Rc;
use std::cell::RefCell;

impl Solution {
    pub fn postorder_traversal(root: Option<Rc<RefCell<TreeNode>>>) -> Vec<i32> {
        fn traverse(node: Rc<RefCell<TreeNode>>, values: &mut Vec<i32>) {
            let node = Rc::try_unwrap(node).unwrap().into_inner();
            
            if let Some(left) = node.left {
                traverse(left, values);
            }
            if let Some(right) = node.right {
                traverse(right, values);
            }
            values.push(node.val);
        }
        let mut values = vec![];
        if let Some(node) = root {
            traverse(node, &mut values);
        }
        values
    }
}